src/components/Api.js

```diff
@@ -0,0 +1,88 @@
+class Api {
+  constructor(options) {
+    // constructor body
+    this._headers = options.headers;
+    this.baseUrl = options.baseUrl;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error ${res.status}`);
+  }
+
+  getInitialCards() {
+    return fetch(this.baseUrl + "/cards", {
+      headers: this._headers,
+    }).then(this._checkResponse);
+  }
+
+  // other methods for working with the API
+
+  loadInfo() {
+    return fetch(this.baseUrl + "/users/me", {
+      method: "GET",
+      avatar:
+        "https://practicum-content.s3.us-west-1.amazonaws.com/frontend-developer/moved_avatar.jpg",
+      cohort: "group-42",
+      headers: this._headers,
+    }).then(this._checkResponse);
+  }
+
+  editProfile(name, about) {
+    return fetch(this.baseUrl + "/users/me", {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        name: name,
+        about: about,
+      }),
+    }).then(this._checkResponse);
+  }
+
+  // pass name and link as argument
+  addNewCard(data) {
+    return fetch(this.baseUrl + "/cards", {
+      method: "POST",
+      headers: this._headers,
+      "Content-Type": "application/json",
```

Comment:
`      "Content-Type": "application/json",
`

This is not needed in the options because `headers: this._headers,` has it inside


**Please, correct it everywhere**